HHS/Head-Start-TTADP
# [Prod] Goal pilot plus additional changes

## Description of change
These are mostly backend changes with a few exceptions. This is what was done:

- Developed a script to create a pilot goal for a select number of participants
- Raised test coverage threshold to 75% on backend
- Added unique ids to buttons across the applications for analytics 
- Implemented an improved icon for draft goals
- Combined the e2e report text test with other filters
- Updated the Elasticsearch return limits from 10 to 2001
- Added e2e test for multiple recipients
- Fixed an incorrect grant number and status display for a case where the same goal for one recipient with two grants was used on 2 different ARs where each grant had it's own AR
- Added support for typescript
- Improved the logging of the audit log db interactions
- Migration to create some missing objectives and goals in multi-recipient scenarios (1065)
- Migration to generate correct goal and objective state transitions, then dedupe some objectives that were created more than once. (1209 & 1243)

TTAHUB-1312 - Fix behavior of sidebar on activity report: if a goal is placed in edit mode, then the sidebar reverts to "in progress," and submission is prevented. 

TTAHU-1329 - Format goal end dates on the backend for the activity report

## How to test
Make sure the tests pass. TTAHUB-1368 can be tested in the UI by following these directions:
https://github.com/HHS/Head-Start-TTADP/pull/1299

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1340
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1047
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1327
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1391
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-872
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1386
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1339
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1368
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1384
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1186
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1312
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1329
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1065
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1209
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1243

## Checklists

### Every PR


- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [n/a] UI review complete

### Production Deploy


### After merge/deploy
